resources/asciidoctor/lib/inline_callout/extension.rb

```diff
@@ -0,0 +1,69 @@
+require 'asciidoctor'
+
+include Asciidoctor
+
+##
+# Enables inline callouts which asciidoc supports but asciidoctor doesn't.
+# Filed as enhancement request at
+# https://github.com/asciidoctor/asciidoctor/issues/3037
+#
+# Usage
+#
+#   POST <1> /_search/scroll <2>
+#
+# NOTE: This isn't an asciidoctor extension so much as a hack. Just including
+# the file causes us to monkey patch its code into asciidoctor. By  default we
+# don't change asciidoctor, but if you set the `inline-callouts` attribute so
+# you need to *ask* for the change in behavior.
+#
+module InlineCallout
+  InlineCalloutScanRx = /\\?<!?(|--)(\d+|\.)\1>/
+  InlineCalloutSourceRx = %r(((?://|#|--|;;) ?)?(\\)?&lt;!?(|--)(\d+|\.)\3&gt;)
+  InlineCalloutSourceRxt = "(\\\\)?&lt;()(\\d+|\\.)&gt;"
+  InlineCalloutSourceRxMap = ::Hash.new {|h, k| h[k] = /(#{::Regexp.escape k} ?)?#{InlineCalloutSourceRxt}/ }
+
+  # Disable VERBOSE so we don't log any warnings. It really isn't great to
+  # have to patch these in like this but it gets the job done and we're looking
+  # to get this into Asciidoctor proper. These methods are basically the same
+  # as the methods in asciidoctor but with new regexes.
+  old_verbose = $VERBOSE
+  $VERBOSE = false
+  Parser.class_eval do
+    def self.catalog_callouts(text, document)
+      found = false
+      autonum = 0
+      callout_rx = (document.attr? 'inline-callouts') ? InlineCalloutScanRx : CalloutScanRx
+      text.scan(callout_rx) {
```

Comment:
Is the body here duplicated from what you are patching over? In particular if `inline-callouts` is _not_ specified, I mean. Point being, in the absence of `inline-callouts` can this delegate to the normal implementation? (I'm probably showing how much ruby I don't really do, here...)

Reply:
I can delegate, yeah. I wasn't partially because I figured this might be what it'd look like to make the change in asciidoctor itself. But it looks like they won't want this change until at least 2.0, which won't look like this anyway. So I can just delegate.